lib/helper/update_manager.dart

```diff
@@ -0,0 +1,139 @@
+    // TODO: maybe also put the check every x hours in the settings
+    if (lastChecked == null || DateTime.now().difference(lastChecked!).inHours > checkEveryHours) {
+      lastChecked = DateTime.now();
+      PreferencesManager.instance.setDateTime("lastChecked", lastChecked!);
+
+      final versionAndUrl = await getVersionAndUrl();
+      if (versionAndUrl?.url != null && await _isNewVersionAvailable(versionAndUrl?.version)) {
+        if (Platform.isAndroid) {
+          // Show notification
+          await _showNotification(versionAndUrl!);
+        } else {
+          // Show dialog
+          showDialog(
+              context: context,
+              builder: (BuildContext c) => simpleDialog.SimpleDialog.create(
+                  context: c,
+                  title: updateNotificationTitle,
+                  content: updateNotificationBody + versionAndUrl!.version.toString(),
+                  okButtonText: "Aktualisieren",
+                  cancelButtonText: "Später",
+                  onSubmitted: () => HelperMethods.openUrl(versionAndUrl.url!)));
+        }
+      }
+    }
+  }
+
+  static Future<void> _showNotification(VersionAndUrl versionAndUrl) async {
+    const AndroidNotificationDetails androidPlatformChannelSpecifics = AndroidNotificationDetails(
+        'updateNotifications', 'Update Benachrichtigungen',
+        channelDescription: 'Benachrichtigungen über neue Updates',
+        importance: Importance.low,
+        priority: Priority.low,
+        ticker: 'SmartHome');
+    const NotificationDetails platformChannelSpecifics = NotificationDetails(android: androidPlatformChannelSpecifics);
+    await flutterLocalNotificationsPlugin.show(notificationId++, updateNotificationTitle,
+        updateNotificationBody + versionAndUrl.version.toString(), platformChannelSpecifics,
+        payload: versionAndUrl.url);
+  }
+
+  static String getVersionString(Version? newVersion) {
+    var v = version.toString();
+
+    if (newVersion != null) {
+      v += " (neue Version " + newVersion.toString() + " verfügbar)";
+    }
+
+    return v;
+  }
+
+  static Future<bool> isNewVersionAvailable() async {
+    final newVersion = await _getNewVersion();
+    return _isNewVersionAvailable(newVersion);
+  }
+
+  static Future<bool> _isNewVersionAvailable(Version? newVersion) async {
+    return newVersion != null && newVersion > version;
+  }
+
+  static Future<VersionAndUrl?> getVersionAndUrl() async {
+    final v = await _getNewVersion();
+    if (v == null) return null;
+
+    final url = await _newVersionUrl();
+
+    return VersionAndUrl(v, url);
+  }
+
+  static Future<Version?> _getNewVersion() {
+    return gitHub.repositories.listTags(repositorySlug).asyncMap((element) {
+      try {
+        return Version.parse(element.name.replaceFirst(versionRegExp, ''));
+      } catch (e) {
+        // TODO: log
+        return null;
+      }
+    }).firstWhere((element) => element != null && element > version);
```

Comment:
```suggestion
    }).firstWhere((element) => (element ?? Version(0, 0, 0)) > version, orElse: () => null);
```

Reply:
You need to supply an orElse if there is the possibility of an element not be found. This is the same as the "First" extension method in c# and this way it's more like "FirstOrDefault"